geopandas/tests/test_cmapplotting.py

```diff
@@ -0,0 +1,240 @@
+import matplotlib as mpl
+
+from shapely.geometry import Point, Polygon, LineString
+
+from geopandas import GeoSeries, GeoDataFrame, plotting
+from geopandas.tests.test_plotting import _check_colors
+
+import pytest
+
+@pytest.fixture(autouse=True)
+def close_figures(request):
+    yield
+    plt.close('all')
+
+try:
+    cycle = mpl.rcParams['axes.prop_cycle'].by_key()
+    MPL_DFT_COLOR = cycle['color'][0]
+except KeyError:
+    MPL_DFT_COLOR = matplotlib.rcParams['axes.color_cycle'][0]
+
+class TestPointPlotting:
+    def setup_method(self):
+        # Intializing the dataframe
+        self.N = 10
+        self.points = GeoSeries(Point(i, i, i) for i in range(self.N))
+
+        values = np.arange(self.N)
+        levels = ['1','1','1','1','1','1','2','2','2','2']
+
+        self.df = GeoDataFrame({'geometry': self.points, 'values': values, 'levels': levels})
+
+    def test_customnorm(self):
+        # test color without additional parameters
+        ax = self.df.plot(column='values')
+        actual_color = ax.collections[0].get_facecolors()
+        cmap = ax.collections[0].get_cmap()
+        expected_color = cmap(np.arange(self.N) / (self.N - 1))
+        np.testing.assert_array_equal(actual_color, expected_color)
+
+        # test color with providing norm
+        norm = mpl.colors.Normalize(vmin = 0, vmax = 12)
+        ax = self.df.plot(column='values', norm = norm)
+        actual_color = ax.collections[0].get_facecolors()
+        cmap = ax.collections[0].get_cmap()
+        expected_color = cmap(norm(list(self.df['values'])))
+        np.testing.assert_array_equal(actual_color, expected_color)
+
+        # test color with providing BoundaryNorm
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.Normalize(vmin = 0, vmax = 12)
+        ax = self.df.plot(column='values', norm = norm)
+        actual_color = ax.collections[0].get_facecolors()
+        cmap = ax.collections[0].get_cmap()
+        expected_color = cmap(norm(list(self.df['values'])))
+        np.testing.assert_array_equal(actual_color, expected_color)
+
+    def test_minmax(self):
+        # test colorbar min,max without additional parameters
+        ax = self.df.plot(column='values', legend=True)
+        colorbar_color = ax.get_figure().axes[1].collections[0].get_clim()
+        point_color = ax.collections[0].get_clim()
+        assert colorbar_color == point_color
+
+        # test colorbar min,max with norm
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.Normalize(vmin = 0, vmax = 12)
+        ax = self.df.plot(column='values', legend=True, norm = norm)
+        colorbar_color = ax.get_figure().axes[1].collections[0].get_clim()
+        point_color = ax.collections[0].get_clim()
+        assert colorbar_color == point_color
+
+        # test color with providing BoundaryNorm
+        cmap = mpl.cm.Greys
+        norm = mpl.colors.BoundaryNorm([1,3,6,8], cmap.N)
+        ax = self.df.plot(column='values', legend=True, norm = norm)
```

Comment:
I would also pass the cmap that you created (and that you use for `norm` to the plot call her (so passing `cmap=cmap`)

Reply:
I agree and I'll make the changes by passing ```cmap = cmap```.